frontend/model/notifications/nativeNotification.js

```diff
@@ -4,50 +4,112 @@ import sbp from '@sbp/sbp'
 // NOTE: since these functions don't modify contract state, it should
 //       be safe to modify them without worrying about version conflicts.
 
-export async function requestNotificationPermission (force: boolean = false): Promise<null | string> {
-  if (typeof Notification === 'undefined') {
-    return null
-  }
+const handler = (statuses: string[]) => {
+  // For some reason, Safari seems to always return `'prompt'` with
+  // `Notification.permission` being correct.
+  const granted = statuses.every(status => status === 'granted') || (statuses.every(status => status === 'prompt') && Notification.permission === 'granted')
+  sbp('state/vuex/commit', 'setNotificationEnabled', granted)
+}
 
-  if (force || Notification.permission === 'default') {
-    try {
-      sbp('state/vuex/commit', 'setNotificationEnabled', await Notification.requestPermission() === 'granted')
-    } catch (e) {
-      console.error('requestNotificationPermission:', e.message)
-      return null
+const fallbackChangeListener = () => {
+  if (!Notification.permission) return
+  let oldValue = Notification.permission
+  handler([oldValue])
+  setInterval(() => {
+    const newValue = Notification.permission
+    if (oldValue !== newValue) {
+      oldValue = newValue
+      handler([oldValue])
     }
-  }
+  }, 100)
```

Comment:
Potential memory leak: interval never cleared

Reply:
This interval isn't meant to be cleared. It's used as a fallback when `navigator.permissions.query` isn't available and is supposed to react "automatically" to permission changes.

Reply:
I see.

Something else: the interval uses `100`. Can you set this to something like `800` or `1000` even? `100` is too quick for something to be happening all of the time that doesn't really benefit much from instantaneous reaction.